### DISCUSSION TITLE
License key validation endpoint

### DISCUSSION THREAD

**Discussion by Brugman** [Make-Lemonade/lemonsqueezy-wp#35] [Ideas]
Can we add a license key validation feature?

I'm testing deploying a WP plugin with LS and the /activate and /deactivate endpoints, as used in the [example plugin](https://github.com/Make-Lemonade/lemonsqueezy-wp-updater-example) work great. A friend gave me the idea to have local installs not count against people's install limit, but do require key validation, so we know it's an active customer.

I browsed the source code and found right besides the /activate and /deactivate endpoints, there's a /validate endpoint. However, it's not the feature I'm asking for: It doesn't take any arguments, so it can't check the license I'm sending. I tried to use it, but it failed as expected. It's got a different purpose.

The vanilla LS API [already has this feature](https://docs.lemonsqueezy.com/guides/tutorials/license-keys#checking-the-validity-of-a-license-key), so it shouldn't be too much effort to implement. Right?

**Comment by fedesib** [comment]
+1 for adding this feature to /validate endpoint which would be very useful to check also if the license you are sending is for the right product. This is specially usefult in case you have more products in your shop and one customer got more than one license.

I also checked the vanilla LS API which has this feature and works perfectly.

Thank you for your time.
